pandas/plotting/_matplotlib/timeseries.py

```diff
@@ -205,7 +205,10 @@ def _get_ax_freq(ax: Axes):
 
 
 def _get_period_alias(freq: timedelta | BaseOffset | str) -> str | None:
-    freqstr = to_offset(freq, is_period=True).rule_code
+    if isinstance(freq, BaseOffset):
+        freqstr = freq.name
+    else:
+        freqstr = to_offset(freq, is_period=True).rule_code
```

Comment:
ok I presume you did this because otherwise the plotting tests fail? I'd suggest leaving a comment in your PR explaining that in the future, else it's very hard to understand why this is being done

Reply:
sorry, I'll leave a comment next time in my PR.

Yes, you are right, without these changes tests for plotting in `pandas/tests/plotting/test_datetimelike.py` and `pandas/tests/plotting/test_series.py` fail.